src/lib/stitching/convection/schema.ts

```diff
@@ -0,0 +1,35 @@
+import { createConvectionLink } from "./link"
+import {
+  makeRemoteExecutableSchema,
+  transformSchema,
+  RenameTypes,
+} from "graphql-tools"
+import { readFileSync } from "fs"
+
+export const executableConvectionSchema = async () => {
+  const convectionLink = createConvectionLink()
+  const convectionTypeDefs = readFileSync("src/data/convection.graphql", "utf8")
+
+  // Setup the default Schema
+  const schema = await makeRemoteExecutableSchema({
+    schema: convectionTypeDefs,
+    link: convectionLink,
+  })
+
+  // Remap the names of certain types from Convection to fit in the larger
+  // metaphysics ecosystem.
+  const remap = {
+    Submission: "ConsignmentSubmission",
+    Category: "SubmissionCategoryAggregation",
+    Asset: "SubmissionCategoryAsset",
+    State: "SubmissionStateAggregation",
+  }
```

Comment:
The ideal is no remap file, right? Or is it an acceptable tradeoff in terms of DX / succinctness within the service?

Reply:
IMO I think a remap is the ideal state, a service can work fine internally with it's own names like `Category` but it could never use that globally. Remapping lets local GQL APIs not have to think globally, that only happens here when something is imported. 

The gravity schema is a good example of whitelisting entirely, I could have dropped everything but the one resolver we want and its dependent types instead of renaming the clashing types.

Reply:
Like with Lewitt’s schema, I think these should be as specific as possible and they should also be consistent, so prefix them all with `ConsignmentSubmission*`?